### DISCUSSION TITLE
How to Mux external Audio with stream

### DISCUSSION THREAD

**Discussion by ErfaNEP1** [streamlink/streamlink#4529] [Q&A] [answered]
hello, I have a problem with downloading streams.
I recently tried to download an HLS stream and figured that it has external audio, and the stream video file itself doesn't have any sound.

```
#EXTM3U
#EXT-AFC:2ef91a11adf0c4d59b6948b2bfe42783
#EXT-X-MEDIA:TYPE=AUDIO,GROUP-ID="audio",LANGUAGE="en",NAME="en",AUTOSELECT=YES, DEFAULT=NO,URI="exampleURL"
#EXT-X-MEDIA:TYPE=AUDIO,GROUP-ID="audio",LANGUAGE="fa",NAME="fa",AUTOSELECT=YES, DEFAULT=YES,URI="exampleURL"
#144p
#EXT-X-STREAM-INF:PROGRAM-ID=1,BANDWIDTH=143843,RESOLUTION=264x148,AUDIO="audio"
exampleURL
```

this is the M3U3 file that I am trying to download.
and as I said, the output file doesn't have any sound.

is there a way to mux one of those audios with my video file in Streamlink?

**Comment by bastimeyer** [comment, marked as answer]
[`--hls-audio-select`](https://streamlink.github.io/cli.html#cmdoption-hls-audio-select) allows you to select external audio tracks added in an HLS variant playlist. This requires FFmpeg to be available in your `PATH` env var, or the path to the FFmpeg executable set via [`--ffmpeg-ffmpeg`](https://streamlink.github.io/cli.html#cmdoption-ffmpeg-ffmpeg), as the two separate streams will need to get remuxed into one output stream via the [`MuxedHLSStream`](https://streamlink.github.io/api.html#muxedhlsstream) implementation.
